### PR TITLE
feat: add authorization server

### PR DESCRIPTION
**Pull request by HPrivakos**

### CODE REVIEW

**Review by menduz — COMMENTED** [review]
How is this new argument different than '--target-content'?

File: src/commands/deploy.ts [hunk 1]
```diff
@@ -184,7 +200,7 @@ export async function main(): Promise<void> {
   )
 
   // Uploading data
-  let catalyst: ContentAPI
+  let catalyst: ContentAPI | CatalystClient
```
Comment (menduz) [inline src/commands/deploy.ts:203]: Would you use two different variables with different types instead? To prevent typing oddities

Comment (HPrivakos) [reply src/commands/deploy.ts:203]: I reverted that, it was to prevent type issues I had while developing it

**Comment by menduz** [comment]
Does it work if we want to specify both the target server and the authentication service? I'm thinking that a `--disable-validations` would simplify things a lot as an alternative implementaiton

**Comment by HPrivakos** [comment]
> Does it work if we want to specify both the target server and the authentication service? I'm thinking that a `--disable-validations` would simplify things a lot as an alternative implementaiton

No, only one can be specified at a time, and we need to check permissions against the server to avoid users without permissions trying to upload scenes against the middleware server

**Comment by menduz** [comment]
I'd suggest to add the `--skip-validations` instead, it seems to be a simpler approach. The content-servers should only expose two endpoints to upload a scene: `POST /entities` and `POST /available-content`.

It would be something like `dcl deploy -t https://my-content-server --skip-validations`

That would also enable people to upload content to *any* content-server compatible API without performing validations.

**Comment by HPrivakos** [comment]
Moved to skip validations and also updated linker-dapp: [https://github.com/decentraland/linker-dapp/pull/12](https://github.com/decentraland/linker-dapp/pull/12)

**Review by menduz — APPROVED** [review]

**Pull request merged**